LayoutTests/inspector/runtime/callFunctionOn-awaitPromise.html

```diff
@@ -0,0 +1,263 @@
+                functionDeclaration: `async function() { return frames[0].iframePromise; }`,
+                awaitPromise: true,
+                returnByValue: true,
+                objectId: thisObject.result.objectId
+            });
+            InspectorTest.expectShallowEqual(result.result.value, {b: 2023}, `Should return the JSON object from iframe.`);
+        },
+    });
+
+    suite.addTestCase({
+        name: "AwaitPromiseRejection",
+        description: "Check that callFunctionOn with awaitPromise=true will await promise rejection.",
+        async test() {
+            let thisObject = await RuntimeAgent.evaluate.invoke({expression: `myObject`});
+            let result = await RuntimeAgent.callFunctionOn.invoke({
+                functionDeclaration: `function() { return new Promise((fulfill, reject) => setTimeout(reject, 10, this)); }`,
+                awaitPromise: true,
+                returnByValue: true,
+                objectId: thisObject.result.objectId
+            });
+            InspectorTest.expectTrue(result.wasThrown, "Should throw an error.");
+            let thrownValue = await RuntimeAgent.callFunctionOn.invoke({
+                functionDeclaration: `function() { return this; }`,
+                awaitPromise: true,
+                returnByValue: true,
+                objectId: result.result.objectId
+            });
+            InspectorTest.expectShallowEqual(thrownValue.result.value, expectedObject, `Should throw the JSON object.`);
+        },
+    });
+
+    suite.addTestCase({
+        name: "AwaitReturnHandlePromiseRejection",
+        description: "Check that callFunctionOn with awaitPromise=true and returnByValue=false will await promise rejection.",
+        async test() {
+            let thisObject = await RuntimeAgent.evaluate.invoke({expression: `myObject`});
+            let result = await RuntimeAgent.callFunctionOn.invoke({
+                functionDeclaration: `function() { return new Promise((fulfill, reject) => setTimeout(reject, 10, this)); }`,
+                awaitPromise: true,
+                returnByValue: false,
+                objectId: thisObject.result.objectId
+            });
+            InspectorTest.expectTrue(result.wasThrown, "Should throw an error.");
+            let preview = await RuntimeAgent.getPreview.invoke({objectId: result.result.objectId});
+            InspectorTest.log(preview);
+        },
+    });
+
+    suite.addTestCase({
+        name: "AwaitPromiseWindowReload",
+        description: "Check that callFunctionOn with awaitPromise=true will receive objected from reloading window.",
+        async test() {
+            let thisObject = await RuntimeAgent.evaluate.invoke({expression: `myObject`});
+            InspectorTest.deferOutputUntilTestPageIsReloaded();
+            let result = await RuntimeAgent.callFunctionOn.invoke({
+                functionDeclaration: `function() { window.location.reload(); return this; }`,
```

Comment:
it seems like `window.location.reload` only actually schedules the reload rather than immediately doing it, so i guess that's why we're still able to get a JSON payload back from the inspected page

for what reason are we testing this?  generally speaking ive found that tests that reload the test page itself are often _very_ flaky, so im a bit hesitant about this

Reply:
Yes, agreed about flakiness, removed the test. It would only make sense in comparison with `awaitPromise` as a separate command anyway.